contracts/erc20-gateway/ERC20Cogateway.sol

```diff
@@ -203,4 +218,103 @@ contract ERC20Cogateway is
+     * @param _feeGasPrice Gas price at which fee will be calculated.
+     * @param _feeGasLimit Gas limit at which fee will be capped.
+     * @param _valueToken ERC20 token address.
+     * @param _depositor Address of depositor on the origin chain.
+     * @param _blockNumber Block number of origin chain against which storage
+                           proof is generated.
+     * @param _rlpParentNodes Storage merkle proof to verify message declaration
+     *                        on the origin chain.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function confirmDeposit(
+        uint256 _amount,
+        address payable _beneficiary,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit,
+        address _valueToken,
+        address _depositor,
+        uint256 _blockNumber,
+        bytes calldata _rlpParentNodes
+     )
+        external
+        returns (bytes32 messageHash_)
+    {
+        uint256 initialGas = gasleft();
+
+        require(
+            _amount != 0,
+            "Deposit amount should be greater than 0."
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be 0."
+        );
+        require(
+            _depositor != address(0),
+            "Depositor address must not be 0."
+        );
+
+        deployUtilityToken(_valueToken);
+
+        uint256 nonce = nonces[msg.sender];
+        nonces[msg.sender] = nonce.add(1);
+
+        messageHash_ = MessageInbox.confirmMessage(
+            ERC20GatewayBase.hashDepositIntent(
+                valueToken,
+                _amount,
+                _beneficiary
+            ),
+            nonce,
+            _feeGasPrice,
+            _feeGasLimit,
+            _depositor,
+            _blockNumber,
+            _rlpParentNodes
+        );
+
+        uint256 feeAmount = ERC20GatewayBase.reward(
+            initialGas.sub(gasleft()),
+            _feeGasPrice,
+            _feeGasLimit
+        );
+
+        uint256 mintAmount = _amount.sub(feeAmount);
+
+        UtilityTokenInterface(address(_valueToken)).mint(msg.sender, feeAmount);
+
+        UtilityTokenInterface(address(_valueToken)).mint(_beneficiary, mintAmount);
+    }
+
+    function deployUtilityToken(address _valueToken) public {
+        address utilityToken = utilityTokens[_valueToken];
+
+        if(utilityToken == address(0)) {
+            bytes memory utilityTokenSetupCalldata = abi.encodeWithSelector(
+                UTILITY_TOKEN_SETUP_CALLPREFIX,
+                "",
+                "",
+                uint8(0),
+                uint256(0),
+                address(this),
+                _valueToken
+            );
+
+            Proxy utilityTokenProxy = createProxy(
+                genesisUtilityTokenMastercopy,
+                utilityTokenSetupCalldata
+            );
+
+            utilityTokens[_valueToken] = address(utilityTokenProxy);
```

Comment:
Return address of utility token contract from this method.

Reply:
Done